### ISSUE TITLE
Make openml-python PEP 561 compliant - Distributing and Packaging Type Information

### ISSUE THREAD

**Issue by eddiebergman** [openml/openml-python#945]
#### Description
I'm building my own python library that uses openml and I'm converting it to be statically typed. I see that the actual library has type hints but it's not recognized by `mypy`. 
 
`mypy : Skipping analyzing 'openml': found module but no type hints or library stubs`

When reading [mypy's docs](https://mypy.readthedocs.io/en/latest/installed_packages.html#making-pep-561-compatible-packages) they give quick instructions on how to do this with [PEP 561 described here](https://www.python.org/dev/peps/pep-0561/).

In summary it is to:
* add a blank file called `py.typed` to the packaged directory
* add the string `"py.typed"` to `package_data` in `setup.py`

I think this would be a nice extra compliance feature for the library to have moving forward and can allow any other integrations or allow anyone else building a library with openml to use their types.

#### Steps/Code to Reproduce
```BASH
> mypy -c 'import openml'      
  
<string>:1: error: Skipping analyzing 'openml': found module but no type hints or library stubs
<string>:1: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```

#### Expected Results
```BASH
> mypy -c 'import openml'

Success: no issues found in 1 source file
```   

#### Versions
Linux-5.8.10-arch1-1-x86_64-with-glibc2.2.5
Python 3.8.5 (default, Sep 21 2020, 12:33:30) 
[GCC 10.2.0]
NumPy 1.19.2
SciPy 1.5.2
Scikit-Learn 0.23.2
OpenML 0.10.2
mypy 0.782

**Comment by mfeurer** [comment]
That would be a great addition. Would you be able to contribute a PR making the required changes and adding `mypy -c 'import openml'` to the unit tests?

**Comment by eddiebergman** [comment]
I'm new to contributing to open-source but I'll give it a go :)

I've read the associated docs for contributing and pull requests, is there a particular folder to put this unit test, seeing as it differs in form from the others or should I make a new folder?

**Comment by mfeurer** [comment]
Thanks a lot :)

No, actually, it would be great if we just test that right after the installation, so you can add the command into this file: https://github.com/openml/openml-python/blob/develop/ci_scripts/install.sh

**Comment by mfeurer** [comment]
Closed via #946
